sbioapputils/app_runner/yaml_automation.py

```diff
@@ -0,0 +1,206 @@
+        #adjusting naming conventions
+        if 'help' in subdict:
+            parameters[key]['tooltip'] = parameters[key].pop('help')
+        if 'choices' in subdict:
+            parameters[key]['options'] = parameters[key].pop('choices')
+        if 'min' in subdict:
+            parameters[key]['min_value'] = parameters[key].pop('min')
+        if 'max' in subdict:
+            parameters[key]['max_value'] = parameters[key].pop('max')
+        #adjusting number formatting
+        if 'type' in subdict:
+            if subdict['type'] in ['int', 'float']:
+                if 'default' in subdict:
+                    subdict['default'] = _attempt_numeric(subdict['default'], subdict['type'])
+                if 'min_value' in subdict:
+                    subdict['min_value'] = _attempt_numeric(subdict['min_value'], subdict['type'])
+                if 'max_value' in subdict:
+                    subdict['max_value'] = _attempt_numeric(subdict['max_value'], subdict['type'])
+                if 'increment' in subdict:
+                    subdict['increment'] = _attempt_numeric(subdict['increment'], subdict['type'])
+        #remove excess quotes from tooltip
+        if 'tooltip' in subdict:
+            raw_tip = subdict['tooltip'].strip("'")
+            subdict['tooltip'] = str(raw_tip)
+        #removing unknown arguments
+        del_list = []
+        for argkey in subdict.keys():
+            if argkey not in allowed_args:
+                del_list.append(argkey)
+        [parameters[key].pop(argkey) for argkey in del_list]
+    return(parameters)
+
+
+def input_yaml_from_args(parameters):
+    
+    input_settings = {}
+    
+    for parameter_dict in parameters.values():
+        if all(k in parameter_dict.keys() for k in ("type","default")):
+            if parameter_dict['type'] in ['path','str']:
+                file_split = parameter_dict['default'].split('.')
+                #if a path, then usually has two parts
+                if len(file_split) == 2:
+                    filename = file_split[0]
+                    fileext = file_split[1]
+                    if fileext in csv_template['allowedFormats']['fileExtensions']:
+                        input_template = csv_template.copy()
+                        input_settings[filename] = input_template
+                    elif fileext in image_template['allowedFormats']['fileExtensions']:
+                        input_template = image_template.copy()
+                        input_settings[filename] = input_template
+                    elif fileext in sc_template['allowedFormats']['fileExtensions']:
+                        input_template = sc_template.copy()
+                        input_settings[filename] = input_template
+                    #commenting out to avoid false positives
+                    #else:
+                    #    input_template = default_template.copy()
+                    #    input_settings[filename] = input_template
```

Comment:
I'll take this out completely. The function is trying to guess if a parameter is a path: if it can find the file extension from list of allowed files I think the guess will be reasonably accurate, but if it only finds an unknown file extension or something that might not even be a file extension I think its better to not include in input_settings.